### PR TITLE
DOC: fixing GL08 errors for pandas.DatetimeIndex.as_unit and pandas.DatetimeIndex.freq

### PR DESCRIPTION
**Pull request by jordan-d-murphy**
All GL08 Errors resolved in the following cases:

1. scripts/validate_docstrings.py --format=actions --errors=GL08 pandas.DatetimeIndex.as_unit
2. scripts/validate_docstrings.py --format=actions --errors=GL08 pandas.DatetimeIndex.freq

_Note: [6232a2e](https://github.com/pandas-dev/pandas/pull/57562/commits/6232a2ed2590fb8d8b43658adc009688cb30dce2) commit message was a typo, this commit is for GL08 errors_

- [x] xref https://github.com/pandas-dev/pandas/issues/57443
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/arrays/datetimelike.py [hunk 1]
```diff
@@ -2038,7 +2038,30 @@ def _validate_dtype(cls, values, dtype):
     def freq(self):
         """
         Return the frequency object if it is set, otherwise None.
-        """
+
+        To learn more about the frequency strings, please see `this link
+        <https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#offset-aliases>`__.
```
Comment (mroeschke) [inline pandas/core/arrays/datetimelike.py:2043]: Can you use ``` :ref:`this link<timeseries.offset_aliases>` ```

Comment (jordan-d-murphy) [reply pandas/core/arrays/datetimelike.py:2043]: you bet! thanks for this, I found several other places with this same link and using the old format, I'll fix this one here in a new commit, and I'll open a separate PR to address the others.

**Comment by jordan-d-murphy** [comment]
pre-commit.ci autofix

File: pandas/core/arrays/datetimelike.py [hunk 2]
```diff
@@ -2151,6 +2174,43 @@ def unit(self) -> str:
...
 
     def as_unit(self, unit: str, round_ok: bool = True) -> Self:
+        """
+        Convert to a dtype with the given unit resolution.
+
+        Parameters
+        ----------
+        unit : {'s', 'ms', 'us', 'ns'}
+        round_ok : bool, default True
+            If False and the conversion requires rounding, raise.
```
Comment (bergnerjonas) [inline pandas/core/arrays/datetimelike.py:2184]:
```suggestion
            If False and the conversion requires rounding, raise ValueError.
```

Comment (jordan-d-murphy) [reply pandas/core/arrays/datetimelike.py]: Thanks for catching this! committed a fix.

**Review by jordan-d-murphy — COMMENTED** [review]
Made a few updates to get the CI checks passing.

**Comment by jordan-d-murphy** [comment]
pre-commit.ci autofix

**Review by bergnerjonas — APPROVED** [review]

**Review by mroeschke — APPROVED** [review]

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jordan-d-murphy

**Pull request merged**